src/Stratis.SmartContracts.Executor.Reflection/CallDataSerializer.cs

```diff
@@ -1,56 +1,65 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using CSharpFunctionalExtensions;
 using NBitcoin;
+using Nethereum.RLP;
 using Stratis.SmartContracts.Core;
 using Stratis.SmartContracts.Executor.Reflection.Serialization;
 
 namespace Stratis.SmartContracts.Executor.Reflection
 {
     public class CallDataSerializer : ICallDataSerializer
     {
-        private readonly IMethodParameterSerializer methodParamSerializer;
+        public const int OpcodeSize = sizeof(byte);
+        public const int VmVersionSize = sizeof(int);
+        public const int GasPriceSize = sizeof(ulong);
+        public const int GasLimitSize = sizeof(ulong);
+        public const int PrefixSize = OpcodeSize + VmVersionSize + GasPriceSize + GasLimitSize;
 
-        private const int intLength = sizeof(int);
+        private readonly IMethodParameterSerializer methodParamSerializer;
+        private readonly IContractPrimitiveSerializer primitiveSerializer;
 
-        public CallDataSerializer(IMethodParameterSerializer methodParameterSerializer)
-        {
-            this.methodParamSerializer = methodParameterSerializer;
+        public CallDataSerializer(IContractPrimitiveSerializer primitiveSerializer)
+        {            
+            this.primitiveSerializer = primitiveSerializer;
+            this.methodParamSerializer = new MethodParameterByteSerializer(primitiveSerializer);
```

Comment:
Normally I'd inject this, but I feel this is OK here because method param serialization could be a responsibility of `CallDataSerializer`. It's only advantageous to keep it separate for testing purposes.

Reply:
Makes sense, it was the responsibility of this prior